resources/js/components/ChatComponent.vue

```diff
@@ -0,0 +1,304 @@
+
+    data() {
+        return {
+            channels: [],
+            userId: null,
+            userName: null,
+            message: null
+        }
+    },
+
+    methods: {
+        joinPublic(event) {
+            let channelName = prompt('Enter the public channel name (e.g. notification)');
+            if (!channelName || channelName.trim().length === 0) {
+                return;
+            }
+            channelName = channelName.trim();
+
+            Echo.channel(channelName)
+                .subscribed(() => {
+                    let channel = {
+                        type: 'public',
+                        channel: channelName,
+                        messages: []
+                    };
+
+                    this.channels.push(channel);
+                    this.setActiveChannel(channel);
+                    this.pushStatusMessage(channel, "Subscribed to public channel " + channelName);
+                })
+                .listenToAll((eventName, data) => {
+                    console.log("Event ::  " + eventName + ", data is ::" + JSON.stringify(data));
+
+                    let channel = this.getChannelByName(channelName, 'public');
+                    this.pushPublicNotification(channel, data)
+                })
+                .error((err) => {
+                    if('statusCode' in err) {
+
+                        if (err.statusCode === 401)
+                            alert("You don't have the access to join this public channel.");
+                        else
+                            alert("An error occurred while trying to join a public channel, check the console for details.");
+                    }
+
+                    console.error(err);
+                });
+        },
+
+        joinPrivate(event) {
+            let channelName = prompt('Enter the private channel name (e.g. room-1, room-2)');
+            if (!channelName || channelName.trim().length === 0) {
+                return;
+            }
+            channelName = channelName.trim();
+
+            Echo.private(channelName)
+                .subscribed(() => {
+                    let channel = {
+                        type: 'private',
+                        channel: channelName,
+                        messages: []
+                    };
+
+                    this.channels.push(channel);
+                    this.setActiveChannel(channel);
+                    this.pushStatusMessage(channel, "Subscribed to private channel " + channelName);
+                })
+                .listenToAll((eventName, data) => {
+                    let channel = this.getChannelByName(channelName, 'private');
+
+                    console.log("Event ::  " + eventName + ", data is ::" + JSON.stringify(data));
+                    if(eventName === '.client-message')
+                        this.pushUserMessage(channel, data.message, data.user);
+                    else
+                        this.pushPublicNotification(channel, data)
```

Comment:
It's better to have a separate `listen` for this.  you can do
echo
listen('public-notification', ()=>)